RecoPixelVertexing/PixelTriplets/plugins/CAHitQuadrupletGeneratorKernels.cu

```diff
@@ -246,23 +250,27 @@ void CAHitQuadrupletGeneratorKernels::launchKernels( // here goes algoparms....
   assert(nhits <= PixelGPUConstants::maxNumberOfHits);
   
   if (earlyFishbone_) {
-    auto blockSize = 128;
+    auto blockSize = 64;
     auto stride = 4;
     auto numberOfBlocks = (nhits + blockSize - 1)/blockSize;
-    numberOfBlocks *=stride;
-  
-    fishbone<<<numberOfBlocks, blockSize, 0, cudaStream>>>(
+    dim3 blks(1,numberOfBlocks,1);
```

Comment:
here and later, and in the kernel code: do we expect any differences using
```c++
dim3 blks(1,numberOfBlocks,1);
dim3 thrs(stride,blockSize,1);
```
or
```c++
dim3 blks(numberOfBlocks,1,1);
dim3 thrs(blockSize,stride,1);
```
assuming the `.x` and `.y` are swapped accordingly inside the kernels ?

Reply:
In fact, do we expect any performance difference using
```
kernel<<<(1, blocks, 1), (stride, size,  1)>>>(...);
```
or
```
kernel<<<blocks, size*stride>>>(..., stride);
```
?

Reply:
Thanks for spitting the PR.

Answer to first question:
According to CUDA doc and examples "x" run faster then "y" so swapping "x" with "y" will NOT achieve the desired result of having the inner loop run in contiguous cuda thread:
The current implementation should be in my intentions equivalent to the hand-made one in terms of thread assignment.

second question:
IN PRINCIPLE the two approaches should be fully equivalent: the use of a 2D grid is clearly more CUDA-style, and does not require the percolation of the stride.
I should have coded directly using the 2D grid.
IN PRACTICE: I cannot exclude a different overhead between the two implementations.
I have simple unit tests/examples

https://github.com/VinInn/ctest/blob/master/cuda/combiHM.cu

https://github.com/VinInn/ctest/blob/master/cuda/combiXY.cu

The hand-made seems a bit faster.

My opinion is that the 2D grid is the way to code it in CUDA: It is surely more easy to understand and maintain.  (is like in C using 1D arrays and computing the offset by hands instead of using a 2D array...)
We could investigate with cuda/nvcc experts: not sure we get anywhere.